src/validator/client/fetcher.ts

```diff
@@ -0,0 +1,272 @@
+/* eslint-disable @typescript-eslint/consistent-type-assertions */
```

Comment:
Ok, stopping here with comments for now because I have to run. More coming after I'm back!